MSPeekCollectionViewDelegateImplementation/Classes/MSPeekCollectionViewDelegateImplementation.swift

```diff
@@ -80,23 +80,32 @@ open class MSPeekCollectionViewDelegateImplementation: NSObject {
         return CGFloat(index) * (itemLength(scrollView) + cellSpacing)
     }
     
-    fileprivate func getNumberOfItemsToScroll(scrollDistance: CGFloat, scrollWidth: CGFloat) -> Int {
+    fileprivate func getNumberOfItemsToScroll(scrollDistance: CGFloat, scrollWidth: CGFloat, velocityInDirection: CGFloat) -> Int {
         var coefficent = 0
-        let safeScrollThreshold = max(scrollThreshold, 0.1)
         
-        switch scrollDistance {
-        case let x where abs(x/safeScrollThreshold) <= 1:
-            coefficent = Int(scrollDistance/safeScrollThreshold)
-        case let x where Int(abs(x/scrollWidth)) == 0:
-            coefficent = max(-1, min(Int(scrollDistance/safeScrollThreshold), 1))
-        default:
-            coefficent = Int(scrollDistance/scrollWidth)
-        }
-        
-        if coefficent > 0 {
-            coefficent = max(minimumItemsToScroll, coefficent)
-        } else if coefficent < 0 {
-            coefficent = min(-minimumItemsToScroll, coefficent)
+        if velocityInDirection > 0 {
+            
+            let safeScrollThreshold = max(scrollThreshold, 0.1)
+            
+            switch scrollDistance {
+            case let x where abs(x/safeScrollThreshold) <= 1:
+                coefficent = Int(scrollDistance/safeScrollThreshold)
+            case let x where Int(abs(x/scrollWidth)) == 0:
+                coefficent = max(-1, min(Int(scrollDistance/safeScrollThreshold), 1))
+            default:
+                coefficent = Int(scrollDistance/scrollWidth)
+            }
+            
+            if coefficent > 0 {
+                coefficent = max(minimumItemsToScroll, coefficent)
+            } else if coefficent < 0 {
+                coefficent = min(-minimumItemsToScroll, coefficent)
+            }
+            
+        } else {
+            
+            coefficent = Int(scrollDistance / ((scrollWidth + cellSpacing) / 2))
```

Comment:
It looks like the number of items to scroll is being ignored when the velocity is negative, is that the intended behavior?

Reply:
Thanks for  reply! @MaherKSantina 

And sorry for my uncautious code. The condition should be `abs(velocity) > 0`.

So my intention is when finger drag ending with no velocity (not negative velocity) , the `destinationIndex` may be judged by (scrollDistance / halfPageWidth). 

Now in the repo, the `destinationIndex` is always judged by (scrollDistance / scrollThreshold), that makes scrolling to next page too easy when the velocity is zero. That means if I drag a little distance (but more than `scrollThreshold`) and stop my finger, the collectionView will scroll to next page.